tests/bluetooth/tester/src/btp_mesh.c

```diff
@@ -899,20 +899,35 @@ static uint8_t priv_node_id_set(const void *cmd, uint16_t cmd_len,
 
 	return BTP_STATUS_SUCCESS;
 }
+#endif
 
+#ifdef CONFIG_BT_MESH_PRIV_BEACON_SRV
 static uint8_t proxy_private_identity_enable(const void *cmd, uint16_t cmd_len,
 					     void *rsp, uint16_t *rsp_len)
 {
+	const struct btp_proxy_priv_identity_cmd *cp = cmd;
+	uint16_t net_idx[CONFIG_BT_MESH_SUBNET_COUNT];
+	ssize_t count;
 	int err;
 
 	LOG_DBG("");
 
-	err = bt_mesh_proxy_private_identity_enable();
-	if (err) {
-		LOG_ERR("Failed to enable proxy private identity (err %d)", err);
+	count = bt_mesh_subnets_get(net_idx, ARRAY_SIZE(net_idx), 0);
+
+	if (count <= 0) {
+		LOG_ERR("No subnet (err:%i)", count);
 		return BTP_STATUS_FAILED;
 	}
 
+	for (int i = 0; i < count; i++) {
+		err = bt_mesh_subnet_priv_node_id_set(net_idx[i], cp->enabled);
```

Comment:
this takes enum bt_mesh_feat_state (which has no explicit values assigned) and not bool so could lead to some confusion (and issues in case eg. enum is reordered)